src/InMemoryEventStore.php

```diff
@@ -37,29 +47,47 @@ public function __construct(ActionEventEmitter $actionEventEmitter)
         $actionEventEmitter->attachListener(self::EVENT_CREATE, function (ActionEvent $event): void {
             $stream = $event->getParam('stream');
 
-            if (isset($this->streams[$stream->streamName()->toString()])) {
+            $streamName = $stream->streamName()->toString();
+
+            if (isset($this->streams[$streamName])
+                || isset($this->cachedStreams[$streamName])
+            ) {
                 return;
             }
 
-            foreach ($stream->streamEvents() as $streamEvent) {
-                $this->streams[$stream->streamName()->toString()]['events'][] = $streamEvent;
+            if ($this->inTransaction) {
+                $this->cachedStreams[$streamName]['events'] = $stream->streamEvents();
+                $this->cachedStreams[$streamName]['metadata'] = $stream->metadata();
+            } else {
+                $this->streams[$streamName]['events'] = $stream->streamEvents();
+                $this->streams[$streamName]['metadata'] = $stream->metadata();
             }
 
-            $this->streams[$stream->streamName()->toString()]['metadata'] = $stream->metadata();
-
             $event->setParam('result', true);
         });
 
         $actionEventEmitter->attachListener(self::EVENT_APPEND_TO, function (ActionEvent $event): void {
-            $streamName = $event->getParam('streamName');
+            $streamName = $event->getParam('streamName')->toString();
             $streamEvents = $event->getParam('streamEvents');
 
-            if (! isset($this->streams[$streamName->toString()])) {
+            if (! isset($this->streams[$streamName])
+                || ! isset($this->streams[$streamName])
```

Comment:
fix via https://github.com/prooph/event-store/commit/65e7a073b0ebf71b591054ead3bab3b1ccc5adcd

thank you!